### ISSUE TITLE
3.5-RC1 - Windows directory bug with project-config

### ISSUE THREAD

**Issue by smcyr** [craftcms/cms#6419]
### Description
No files are written inside the folders of config/project. If I debug the code, Craft tries to write something like that:
`C:\Projects\craft-cms-3\config\project\sections/a3c97649-f850-446b-ae56-7c2fdef07145.yaml`
So the directory separator should be fixed for Windows

### Steps to reproduce

Try to upgrade to 3.5, add a field, section or anything and many bugs appear and no files are written inside the project folder except for project.yaml and the directories.

### Additional info

Windows 10 / IIS
- Craft version: 3.5-RC1
- PHP version: 7.3

**Comment by angrybrad** [comment]
Both `/` and `\` are valid directory separators on Windows, so that shouldn't be the issue.

Are you sure whatever user account the IIS AppPool is running as has permissions to write to `C:\Projects\craft-cms-3\config\project`?

**Comment by smcyr** [comment]
Yes, it can write to the folder, because if I change the forward slashs (/) for `DIRECTORY_SEPARATOR` inside the `splitConfigIntoComponentsInternal` method, it's able to write the files. Do you have a Windows machine to test?

**Comment by angrybrad** [comment]
> Do you have a Windows machine to test?

Yup, just did a fresh install or Craft 3.5-RC1 on Windows 10 using native IIS/MySQL and the files are created as expected:

![Annotation 2020-07-21 134857](https://user-images.githubusercontent.com/61869/88105658-1ac70580-cb59-11ea-9d20-5329f712ab84.jpg)

Anything odd about your setup?  Are you running some non-NTFS filesystem perhaps?

**Comment by angrybrad** [comment]
Also, writing those files out should always go through `FileHelper::writeToFile()` https://github.com/craftcms/cms/blob/3.5/src/services/ProjectConfig.php#L1575

The very first thing it does is call normalizePath: https://github.com/craftcms/cms/blob/develop/src/helpers/FileHelper.php#L344

Which ensures directory separators are consistent across platforms: https://github.com/craftcms/cms/blob/develop/src/helpers/FileHelper.php#L42-L55

**Comment by smcyr** [comment]
I a debug inside the method `writeToFile`, the file is created but is then deleted after. 
It's this part that creates the problem:
```
$basePathLength = strlen($basePath);
            foreach ($this->_findConfigFiles($basePath) as $file) {
                $path = substr($file, $basePathLength + 1);
                if (!isset($config[$path])) {
                    FileHelper::unlink($file);
                }
            }
```
The `$config` variable has forward slashs and the variable $path is the inverse, so it unlink all the files.
And I'm on NTFS and I use PHP with FastCGI.

**Comment by smcyr** [comment]
![Annotation 2020-07-21 170845](https://user-images.githubusercontent.com/664920/88107379-fda03000-cb74-11ea-86be-ebc294719a6c.jpg)

**Comment by angrybrad** [comment]
Got it!  Will push a fix up shortly and thanks for the debugging help!

**Comment by brandonkelly** [comment]
We just released Craft 3.5 RC1.1 with the fix for this. Thanks again for reporting/investigating!

**Comment by smcyr** [comment]
Thanks for the fix! It works now, except that there is now a new related bug: https://github.com/craftcms/cms/issues/6425
